engine/access/state_stream/filter.go

```diff
@@ -139,6 +221,168 @@ func (f *EventFilter) Match(event flow.Event) bool {
 	return false
 }
 
+// matchFieldFilter checks if the provided event matches the field filters defined in the event filter.
+//
+// This method is called internally by the EventFilter's Match method to determine if an event matches the field filters criteria.
+// It compares the field values of the event with the target values specified in the field filters.
+// If the event matches all field filters, it returns true; otherwise, it returns false.
+// If there is an error decoding the event payload or if the event does not contain the expected fields, it returns false.
+func (f *EventFilter) matchFieldFilter(event *flow.Event) bool {
+	fieldFilters, ok := f.EventFieldFilters[event.Type]
+	if !ok {
+		return false // no filter for this event
+	}
+
+	if len(fieldFilters) == 0 {
+		return true // empty list always matches
+	}
+
+	fields, fieldValues, err := getEventFields(event)
+	if err != nil {
+		return false
+	}
+
+	for i, field := range fields {
+		filters, ok := fieldFilters[field.Identifier]
+		if !ok {
+			continue // no filter for this field
+		}
+
+		fieldValue := fieldValues[i].String()
+		if _, ok := filters[fieldValue]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+// AccountStatusFilter defines a specific filter for account statuses.
+// It embeds the EventFilter type to inherit its functionality.
+type AccountStatusFilter struct {
+	*EventFilter
+}
+
+func NewAccountStatusFilter(
+	config EventFilterConfig,
+	chain flow.Chain,
+	eventTypes []string,
+	accountAddresses []string,
+) (AccountStatusFilter, error) {
+	filterEventTypes, err := GetCoreEventTypes(eventTypes)
```

Comment:
I think you could simplify this. 
1. If `eventTypes` is empty, use the `defaultCoreEvents`
2. If `accountAddresses` is empty, pass `eventTypes` as the `EventTypes` arg to `NewEventFilter`. this will make it match on any instance of those events.
3. If `accountAddresses` is not empty, iterate through the accounts and list of events and call `addCoreEventFieldFilter`
4. If an error is returned from `addCoreEventFieldFilter`, return an error since an invalid event type was passed